### DISCUSSION TITLE
Can Supabase subscribe to logical replication? or alt method to continuously update Supabase?

### DISCUSSION THREAD

**Discussion by unibum** [supabase/supabase#29977] [Questions]
Hi,
Looking for a way to get 3rd party Postgresql files to be continuously updated into Supabase.
I see supabase can use logical replication for alternative Db to subscribe to supabase, but no mention if supabase can subscribe to the alt database?
I also see pgloader may be another option.
Does anyone have anything working to solve the above? e.g. daily update (full replacement of data) into Supabase. looking for something automated/schedule that doesn't need manual running each day.

**Comment by Rubenburdin** [comment]
Hey unibum,
Good question, Supabase can act as a logical replication publisher (so external systems can subscribe to it), but acting as a subscriber is trickier. Creating a subscription with CREATE SUBSCRIPTION requires replication privileges that Supabase's managed environment doesn't expose to users. So native logical replication inbound isn't really an option on their platform today.
For the "daily full replacement" approach you mentioned, you could cobble something together with postgres_fdw + pg_cron. The idea is: set up a foreign data wrapper connection to your external Postgres, then schedule a cron job that truncates your local table and does an INSERT INTO ... SELECT * FROM foreign_table. Supabase supports both extensions. Docs for pg_cron are here (https://supabase.com/docs/guides/database/extensions/pg_cron). It works, but it's a full table swap each time - not incremental, and you have to handle the FDW connection credentials carefully.
pgloader is more of a migration tool than a continuous sync solution. You'd have to trigger it externally on a schedule, and it doesn't track changes - just bulk loads.
If you need something closer to real-time or incremental updates without building custom infra, external sync tools are probably the cleaner path. Full disclosure, I work on [Stacksync ](https://www.stacksync.com/)- we handle exactly this kind of Postgres-to-Postgres sync (including Supabase) with CDC so you're not doing full table replacements every time. Might be worth a look depending on your data volume.
